src/cts/src/TritonCTS.cpp

```diff
@@ -517,28 +523,152 @@ std::string TritonCTS::selectRootBuffer(std::vector<std::string>& bufferVector)
+  float rootWireLength = static_cast<float>(std::max(width, height))
+                         / block->getDbUnitsPerMicron();
+  sta::Corner* corner = openSta_->cmdCorner();
+  float rootWireCap
+      = resizer_->wireClkCapacitance(corner) * 1e-6 * rootWireLength;
+  // clang-format off
+  debugPrint(logger_, CTS, "buffering", 2, "core width:{}, core height:{}, "
+             "rootWL:{:0.2e}, rootWireCap:{:0.2e}, clock cap/micron:{:0.2e}", width, height,
+             rootWireLength, rootWireCap,
+             resizer_->wireClkCapacitance(corner) * 1e-6);
+  // clang-format on
+
+  // pick the smallest buffer that can drive root wire cap
+  // if no such buffer exists, pick one that has the largest max cap
+  float bestArea = std::numeric_limits<float>::max();
+  float bestCap = 0.0;
+  std::string rootBuf, nextBestBuf;
   for (const std::string& name : bufferVector) {
     odb::dbMaster* master = db_->findMaster(name.c_str());
     sta::Cell* masterCell = network_->dbToSta(master);
     sta::LibertyCell* libCell = network_->libertyCell(masterCell);
     sta::LibertyPort *in, *out;
     libCell->bufferPorts(in, out);
-    if (out->driveResistance() < bestRes) {
-      bestRes = out->driveResistance();
+    float area = libCell->area();
+    float maxCap = 0.0;
+    bool maxCapExists = false;
+    out->capacitanceLimit(sta::MinMax::max(), maxCap, maxCapExists);
+    if (maxCapExists && maxCap > rootWireCap && area < bestArea) {
       rootBuf = name;
+      bestArea = area;
+    }
+    if (maxCap > bestCap) {
+      nextBestBuf = name;
+      bestCap = maxCap;
+    }
+    // clang-format off
+    if (maxCapExists) {
+      debugPrint(logger_, CTS, "buffering", 2, "{} has max cap:{:0.2e}", name, maxCap);
     }
   }
 
+  if (rootBuf.empty()) {
+    rootBuf = nextBestBuf;
+  }
   // clang-format off
-  debugPrint(logger_, CTS, "buffering", 4, "{} has been selected as root buffer",
-             rootBuf);
+  debugPrint(logger_, CTS, "buffering", 1, "{} has been selected as root "
+             "buffer to drive root wire cap of {:0.2e}", rootBuf, rootWireCap);
   // clang-format on
   return rootBuf;
 }
 
+void TritonCTS::setSinkBuffer(const char* buffers)
+{
+  std::stringstream ss(buffers);
+  std::istream_iterator<std::string> begin(ss);
+  std::istream_iterator<std::string> end;
+  std::vector<std::string> bufferVector(begin, end);
+  std::string sinkBuffer = selectSinkBuffer(bufferVector);
+  options_->setSinkBuffer(sinkBuffer);
+}
+
+std::string TritonCTS::selectSinkBuffer(std::vector<std::string>& bufferVector)
+{
+  // if -sink_clustering_buf is not specified, choose from the buffer list
+  if (bufferVector.empty()) {
+    bufferVector = options_->getBufferList();
+  }
+
+  if (bufferVector.size() == 1) {
+    return bufferVector.front();
+  }
+
+  // estimate wire cap for sink buffer
+  // assume sink buffer needs to drive clk buffers at two far ends of chip
+  // at midpoint (pin caps can't be estimated because sinks have not been
```

Comment:
Wouldn't a sink cluster be a lot smaller than this?

Reply:
Yes, I'm over-estimating because I don't have pin caps yet.